### PR TITLE
feat(ngrx): Added ngrx stuff for planner

### PR DESCRIPTION
**Pull request by sanbornsen**
This PR is about initiating an empty parent state for ```planner``` or any other application will be using ngrx in future. The ngrx work of planner is getting to an end and once it goes to production it'll need this support from the platform. 

This is not a breaking change for the platform.

### CODE REVIEW

**Comment by fabric8cd** [comment]
@sanbornsen fabric8/fabric8-ui:SNAPSHOT-PR-2570-1 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2570-fabric8-ui.badger.fabric8.io

File: package.json [hunk 1]
```diff
@@ -111,6 +111,9 @@
     "@angularclass/hmr": "2.1.3",
     "@angularclass/hmr-loader": "3.0.4",
     "@thisissoon/angular-inviewport": "1.3.2",
+    "@ngrx/effects": "^4.1.1",
```
Comment (joshuawilson) [inline package.json]: please remove the auto-upgrade ^

**Comment by joshuawilson** [comment]
This uses Angular 5.2. I'm planning on updating Angular and then we can include ngrx 5.x.

What aspect of the planner requires ngrx?

**Comment by sanbornsen** [comment]
True! We are also currently developing on ngrx 4.x in the planner. Almost everything is being shifted to faster and better version built on top of ngrx and redux technology. The work is under process for quite some time now. In another week or so it'll be completed. Here is the forked branch - https://github.com/sanbornsen/fabric8-planner/tree/ngrx/store.
Once this goes into the master, the planner would expect these changes in the platform to feel like home.

**Comment by fabric8cd** [comment]
@sanbornsen fabric8/fabric8-ui:SNAPSHOT-PR-2570-2 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2570-fabric8-ui.badger.fabric8.io

File: src/app/app.module.ts [hunk 2]
```diff
@@ -160,7 +164,10 @@ export type StoreType = {
     PatternFlyNgModule,
     StatusListModule,
     // AppRoutingModule must appear last
-    AppRoutingModule
+    AppRoutingModule,
```
Comment (joshuawilson) [inline src/app/app.module.ts]: **AppRoutingModule must appear last**

File: package.json [hunk 3]
```diff
@@ -111,6 +111,9 @@
     "@angularclass/hmr": "2.1.3",
     "@angularclass/hmr-loader": "3.0.4",
     "@thisissoon/angular-inviewport": "1.3.2",
+    "@ngrx/effects": "4.1.1",
```
Comment (joshuawilson) [inline package.json]: This list is alphabetical, please move it above @thisissoon/angular-inviewport

File: src/app/app.module.ts [hunk 4]
```diff
@@ -113,6 +113,10 @@ import { RavenExceptionHandler } from './shared/exception.handler';
 import { ForgeWizardModule } from './space/forge-wizard/forge-wizard.module';
 
 
+// ngrx stuff
+import { EffectsModule } from '@ngrx/effects';
```
Comment (joshuawilson) [inline src/app/app.module.ts]: The imports are in 3 sections. (all the files should be following this layout)
Angular, external, then internal. Each section is alphabetical. Please move this to about line 11.

**Comment by fabric8cd** [comment]
@sanbornsen fabric8/fabric8-ui:SNAPSHOT-PR-2570-3 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2570-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@sanbornsen fabric8/fabric8-ui:SNAPSHOT-PR-2570-4 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2570-fabric8-ui.badger.fabric8.io

**Pull request merged**